src/segments/sitecore.go

```diff
@@ -0,0 +1,96 @@
+package segments
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+)
+
+type Sitecore struct {
+	props properties.Properties
+	env   platform.Environment
+
+	EnvironmentName    string
+	IsCloudEnvironment bool
+	EnvironmentIcon    string
+}
+
+type SitecoreConfig struct {
+	Endpoints struct {
+		Default struct {
+			Ref        string `json:"ref"`
+			AllowWrite bool   `json:"allowWrite"`
+			Host       string `json:"host"`
+			Variables  struct {
+			} `json:"variables"`
+		} `json:"default"`
+	} `json:"endpoints"`
+}
+
+const (
+	CloudIcon properties.Property = "cloud_icon"
+	OtherIcon properties.Property = "other_icon"
+)
+
+func (s *Sitecore) Enabled() bool {
+	return s.shouldDisplay()
+}
+
+func (s *Sitecore) Template() string {
+	return " {{ if .PackageManagerIcon }} {{ .PackageManagerIcon }} {{ end }} {{ .EnvironmentName }} "
+}
+
+func (s *Sitecore) Init(props properties.Properties, env platform.Environment) {
+	s.props = props
+	s.env = env
+}
+
+func (s *Sitecore) shouldDisplay() bool {
+	sitecoreDir, err := s.env.HasParentFilePath(".sitecore")
+	if err != nil {
+		return false
+	}
+
+	if !sitecoreDir.IsDir {
+		return false
+	}
+
+	if !s.env.HasFilesInDir(sitecoreDir.Path, "user.json") {
+		return false
+	}
+
+	sitecoreConfigFile := filepath.Join(sitecoreDir.Path, "user.json")
+
+	if len(sitecoreConfigFile) == 0 {
+		return false
+	}
+
+	content := s.FileContentWithoutBom(sitecoreConfigFile)
+
+	var config SitecoreConfig
+	if err := json.Unmarshal([]byte(content), &config); err != nil {
+		return false
+	}
+
+	// sitecore xm cloud always has sitecorecloud.io as domain
+	s.IsCloudEnvironment = strings.Contains(config.Endpoints.Default.Host, "sitecorecloud.io")
+
+	if s.IsCloudEnvironment {
```

Comment:
I would do this in the template: `{{ if . IsCloudEnvironment }}\uf65e{{ else }}\uf98a{{ end }}`